### ISSUE TITLE
WebSockets for the REST-API to stream updates/changes into the Browser in real-time 

### ISSUE THREAD

**Issue by JohannesLichtenberger** [sirixdb/sirix#286]
This feature could be used by visualizations in the front-end for instance.

**Comment by parameswaranmarimuthu** [comment]
Hi @JohannesLichtenberger could you please provide more details on this? I am can start on this.

**Comment by JohannesLichtenberger** [comment]
I think it would be great if updates to the resources, that is via `XmlNodeTrx` and the `JsonNodeTrx` could be streamed into the Browser for instance. That is someone should be able to subscribe to resource changes within a database (topic might be database + resource name). I think we might be able to extend the HTTP-server (sirix-rest-api) for instance. Or I'm not sure if it makes sense to develop a second, simple server with Quarkus.io. Authentication should also be done via Keycloak / OAuth2.

**Comment by JanBucko** [comment]
Hello, is this issue still free? I would like to take a look.

**Comment by JohannesLichtenberger** [comment]
@JanBucko, sure it's still open and no one assigned. Maybe we should make it optional in the `rest-api` bundle.

**Comment by JanBucko** [comment]
I'm glad to hear that. I will try to get a little bit familiar with the project, then I will see what can I do may I turn on you in case of any issues ?

**Comment by JohannesLichtenberger** [comment]
Yes sure, keep asking here, in our Slack channel or in the Forum. I think the forum might also be a good place for general discussions :+1:

**Comment by JanBucko** [comment]
Hello, i am having a little problem running things locally, is there any guide for building the project please ?

**Comment by JohannesLichtenberger** [comment]
Hi Jan, have a look at https://sirix.io/rest-api.html#api-description

I hope that helps :)

**Comment by JanBucko** [comment]
I am giving up on this issue.

**Comment by JohannesLichtenberger** [comment]
So, you didn't get the HTTP-Server up-and-running? That's a shame. I'd love to help, for sure. Can you give me a hint of what the problem is for other users?

**Comment by JanBucko** [comment]
Hello, i didnt manage to get even the database core examples running :)

**Comment by JohannesLichtenberger** [comment]
What was the stacktrace of the throwable after running with storage type file?

**Comment by JanBucko** [comment]
I have described the whole problem in your comunity forum :) Yor suggestion to change storage type to FILE worked but I came across linkage error when I was trying to launch the main method of example codes as I described, perhaps I have done something terribly wrong. Nevertheless I am currently contributing to some other projects, but when I'll have those issues done perhaps I'll gladly give it another try :) until now I was working only with maven not with graddle, so I am almost It was purely my mistake it didnt work.
